clarity/src/vm/analysis/type_checker/mod.rs

```diff
@@ -947,8 +1444,13 @@ impl<'a, 'b> TypeChecker<'a, 'b> {
                         self,
                         trait_type_size(&trait_signature)?,
                     )?;
-                    self.contract_context
-                        .add_trait(trait_name, trait_signature)?;
+                    if self.clarity_version < ClarityVersion::Clarity2 {
```

Comment:
I see this feature gate repeated a few times.  Might it make more sense to have a single method for adding a trait that takes the trait signature and clarity version as arguments, and carries out the feature-gated calls to either `add_trait()` or `add_defined_trait()`?

Reply:
Refactored in c6eaf08cb